provider/env_test.go

```diff
@@ -0,0 +1,71 @@
+package provider_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/coder/terraform-provider-coder/provider"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestEnv(t *testing.T) {
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]*schema.Provider{
+			"coder": provider.New(),
+		},
+		IsUnitTest: true,
+		Steps: []resource.TestStep{{
+			Config: `
+			provider "coder" {
+			}
+			resource "coder_env" "example" {
+				agent_id = "king"
+				name = "MESSAGE"
+				value = "Believe in yourself and there will come a day when others will have no choice but to believe with you."
+			}
+			`,
+			Check: func(state *terraform.State) error {
+				require.Len(t, state.Modules, 1)
+				require.Len(t, state.Modules[0].Resources, 1)
+				script := state.Modules[0].Resources["coder_env.example"]
+				require.NotNil(t, script)
+				t.Logf("script attributes: %#v", script.Primary.Attributes)
+				for key, expected := range map[string]string{
+					"agent_id": "king",
+					"name":     "MESSAGE",
+					"value":    "Believe in yourself and there will come a day when others will have no choice but to believe with you.",
+				} {
+					require.Equal(t, expected, script.Primary.Attributes[key])
+				}
+				return nil
+			},
+		}},
+	})
+}
+
+func TestEnvNoName(t *testing.T) {
```

Comment:
For completion, you need one more test, when the ENV variable is clear (no value).

Reply:
I can add it, but what behavior specifically did you mean we should test/validate?